### PR TITLE
Adding builds files to the test projects

### PR DESCRIPTION
**Pull request by sokket**
Adding builds files to the test projects so we will only build a test project on the platform (and only with the configuration) that it should build for.

cc @weshaggard

### CODE REVIEW

**Comment by weshaggard** [comment]
@sokket As we talked about last time you shouldn't include TargetGroup configurations in the builds file right now as the tests currently don't support anything beyond the default empty TargetGroup. So please exclude any configurations that define a TargetGroup from the builds files.

**Comment by sokket** [comment]
Sorry about that, I changed my script and it included the TargetGroup nodes...I'll fix it now

**Comment by ellismg** [comment]
Something goofy is going on.  The Winodws_NT legs failed because they were running at least some tests that [were marked as AnyUnix](http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/windows_nt_release_prtest/3029/testReport/junit/(root)/TextInfoTests/EnUsPosixIsNotATurkishCasingLocale/).

**Comment by sokket** [comment]
@ellismg it looks like there isn't any xunit filtering done per-OS...the ZipFile tests had the same problem:

CoreRun.exe xunit.console.netcore.exe System.IO.Compression.ZipFile.Tests.dll  -xml testResults.xml -notrait Benchmark=true  -notrait category=OuterLoop -notrait category=failing 


@weshaggard what generates the xunit command to run? It looks like we need to add the ```-notrait category=<os>``` trait

**Comment by ellismg** [comment]
> @weshaggard what generates the xunit command to run? It looks like we need to add the -notrait category=<os> trait

It's handled by [tests.targets](https://github.com/dotnet/buildtools/blob/13b0cacd70f91e7358f80ca474f87b9b2c7b1058/src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets#L62) in build tools.

**Comment by ellismg** [comment]
`run-test.sh` will also need to be updated since it looks like the paths of stuff in the output folder are changing, and previously it assumed that there would be just one OS folder that all tests went into.

**Comment by sokket** [comment]
@ellismg Thanks for pointing me to that. Maybe that file should be changed to filter based on the OS that it is running on instead of the OSGroup parameter?

**Comment by ellismg** [comment]
> Maybe that file should be changed to filter based on the OS that it is running on instead of the OSGroup parameter?

The only issue with that is I don't believe today we have a property that describes the OS your are running on.  AFAIK we only distinguish between "Windows" and "Unix" for the host OS, so we'd need to add something and then plumb things through.

**Comment by sokket** [comment]
Makes sense... @weshaggard any thoughts on if this will be the right approach?

**Comment by sokket** [comment]
@weshaggard The problem with any of the variables I found is that they do not match up with our test case filters [seen here](https://github.com/dotnet/buildtools/blob/13b0cacd70f91e7358f80ca474f87b9b2c7b1058/src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets#L62)...for instance, we turn FreeBSD distros into Ubuntu runtime but have BSD test filters. I think we will need to plumb something fully through the system to get the right traits

**Comment by sokket** [comment]
Adding some logging to check the state of MSBuild variables on all the platforms...

**Comment by sokket** [comment]
Windows failure is #6732, will retest since it shouldn't be a 100% repro

**Comment by sokket** [comment]
@chcosta FYI

File: dir.targets [hunk 1]
```diff
@@ -28,6 +28,15 @@
...
     <PortableNuGetMode>true</PortableNuGetMode>
+
+    <!--Start with NonUnixTests if we aren't on Windows and overwrite with a specific OS, if we know what it is-->
+    <TestFilterCategory Condition="'$(OS)' != 'Windows_NT'">-notrait category=nonunixtests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'OSX'">-notrait category=nonosxtests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'Windows_NT'">-notrait category=nonwindowstests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'Linux'">-notrait category=nonlinuxtests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'FreeBSD'">-notrait category=nonfreebsdtests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'NetBSD'">-notrait category=nonnetbsdtests</TestFilterCategory>
+    <TestCommandLine>$(TestCommandLine) $(TestFilterCategory)</TestCommandLine>
```
Comment (chcosta) [inline dir.targets]: I'm grabbing a copy of this code change to dir.targets, as-is, for a change I am currently working on.  I'll be sure to cc you on the PR so that you are aware of any conflicts if you make additional changes.  This change is simpler than the build tools change I was considering, only because coordination between buildtools and corefx won't be required.  Is there a plan to move this into BuildTools?  This seems somewhat redundant and likely to cause conflicts at some point given that these are additive and keying off of different properties...

https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets#L60

Comment (sokket) [inline dir.targets]: When I last spoke with @weshaggard he mentioned that he usually puts changes like this into CoreFx to vet it first and then moves it over to build tools after.

Comment (weshaggard) [inline dir.targets]: Yes I'm cool with staging this in corefx and then prompting to buildtools.

As for the specific change I'm not entirely sure $(OS) is the right way to go here. Msbuild will currently only set it to Unix, OSX, or Windows (https://github.com/radical/msbuild/blob/fix-isosx/src/Shared/NativeMethodsShared.cs#L534). I think we probably should start passing it through the builds file as TestOSGroup or something like that which we can default to the OSGroup passed in at the root or we can start explicitly passing it.

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
@dotnet-bot test Innerloop Windows_NT Release Build and Test

**Comment by sokket** [comment]
Some infrastructure failures, sent emails about them. Will retest

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
Machines went offline during the build...retrying

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by mmitche** [comment]
Stand by, I'm restarting the system with an update or two

**Comment by sokket** [comment]
@mmitche cool, let me know when you've completed the update so I can re-test?

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
@weshaggard Things are green! Can you take another look please?

File: dir.targets [hunk 2]
```diff
@@ -28,5 +28,15 @@
     <TargetFrameworkProfile></TargetFrameworkProfile>
     <!-- We set this property to avoid MSBuild errors regarding not setting TargetFrameworkProfile (see above line) -->
     <PortableNuGetMode>true</PortableNuGetMode>
+
```
Comment (weshaggard) [inline dir.targets]: This needs to be reconciled with @Priya91 recent BuildTools PR https://github.com/dotnet/buildtools/pull/564

File: src/Microsoft.VisualBasic/tests/Microsoft.VisualBasic.Tests.csproj [hunk 3]
```diff
@@ -5,6 +5,7 @@
     <ProjectGuid>{82B54697-0251-47A1-8546-FC507D0F3B08}</ProjectGuid>
     <AssemblyName>Microsoft.VisualBasic.Tests</AssemblyName>
     <RootNamespace>Microsoft.VisualBasic.Tests</RootNamespace>
+    <UnsupportedPlatforms>Linux;NetBSD;OSX</UnsupportedPlatforms>
```
Comment (weshaggard) [inline src/Microsoft.VisualBasic/tests/Microsoft.VisualBasic.Tests.csproj]: Is there a reason these still need to be specified?

File: src/Microsoft.Win32.Registry.AccessControl/pkg/Microsoft.Win32.Registry.AccessControl.builds [hunk 4]
```diff
@@ -2,7 +2,9 @@
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <ItemGroup>
-    <Project Include="Microsoft.Win32.Registry.AccessControl.pkgproj" />
```
Comment (weshaggard) [inline src/Microsoft.Win32.Registry.AccessControl/pkg/Microsoft.Win32.Registry.AccessControl.builds]: Why did you need to change the pkg builds file?

File: src/System.Data.SqlClient/tests/ManualTests/System.Data.SqlClient.ManualTesting.Tests.csproj [hunk 5]
```diff
@@ -14,6 +14,7 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Windows_Release|AnyCPU'" />
   <PropertyGroup Condition=" '$(TargetsWindows)' != 'true' ">
     <DefineConstants>$(DefineConstants);MANAGED_SNI</DefineConstants>
+    <UnsupportedPlatforms>Linux;OSX</UnsupportedPlatforms>
```
Comment (weshaggard) [inline src/System.Data.SqlClient/tests/ManualTests/System.Data.SqlClient.ManualTesting.Tests.csproj]: Why are these needed?

File: src/System.Data.SqlClient/tests/FunctionalTests/System.Data.SqlClient.Tests.csproj [hunk 6]
```diff
@@ -9,6 +9,7 @@
     <OutputType>Library</OutputType>
     <RootNamespace>System.Data.SqlClient.Tests</RootNamespace>
     <AssemblyName>System.Data.SqlClient.Tests</AssemblyName>
+    <UnsupportedPlatforms>Linux;OSX</UnsupportedPlatforms>
```
Comment (weshaggard) [inline src/System.Data.SqlClient/tests/FunctionalTests/System.Data.SqlClient.Tests.csproj]: Why are these needed?

File: src/System.Console/tests/System.Console.Tests.builds [hunk 7]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <ItemGroup>
```
Comment (weshaggard) [inline src/System.Console/tests/System.Console.Tests.builds]: Is there a reason this cannot be AnyOS (i.e. empty OSGroup)?

File: src/Microsoft.VisualBasic/tests/Microsoft.VisualBasic.Tests.csproj (hunk 3 above)
Comment (sokket) [inline src/Microsoft.VisualBasic/tests/Microsoft.VisualBasic.Tests.csproj]: So these will still need to be specified until the run-test.sh script is fixed. The run-test script will check and if there is a csproj file without a test dll then it will throw an error; however, if there is an UnsupportedPlatform tag then it will not error out (by design). The changes to fix run-test to work with missing binaries is out-of-scope for this change

File: src/System.Data.SqlClient/tests/System.Data.SqlClient.Tests.builds [hunk 8]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <ItemGroup>
+    <Project Include="FunctionalTests\System.Data.SqlClient.Tests.csproj" />
+    <Project Include="ManualTests\System.Data.SqlClient.ManualTesting.Tests.csproj" />
```
Comment (weshaggard) [inline src/System.Data.SqlClient/tests/System.Data.SqlClient.Tests.builds]: If these aren't supported on some platforms, which appears to be the case because you added UnsupportedPlatforms in their projects why not expand them to only include the supported configurations here?

File: src/System.IO.FileSystem.AccessControl/pkg/System.IO.FileSystem.AccessControl.builds [hunk 9]
```diff
@@ -2,7 +2,9 @@
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <ItemGroup>
-    <Project Include="System.IO.FileSystem.AccessControl.pkgproj" />
```
Comment (weshaggard) [inline src/System.IO.FileSystem.AccessControl/pkg/System.IO.FileSystem.AccessControl.builds]: Why did you update this pkg builds file?

File: src/Microsoft.Win32.Registry.AccessControl/pkg/Microsoft.Win32.Registry.AccessControl.builds (hunk 4 above)
Comment (sokket) [inline src/Microsoft.Win32.Registry.AccessControl/pkg/Microsoft.Win32.Registry.AccessControl.builds]: I got a bin-clash error without this change

File: src/System.Data.SqlClient/tests/ManualTests/System.Data.SqlClient.ManualTesting.Tests.csproj (hunk 5 above)
Comment (sokket) [inline src/System.Data.SqlClient/tests/ManualTests/System.Data.SqlClient.ManualTesting.Tests.csproj]: See note above IRT UnsupportedPlatform

File: src/System.Data.SqlClient/tests/FunctionalTests/System.Data.SqlClient.Tests.csproj (hunk 6 above)
Comment (sokket) [inline src/System.Data.SqlClient/tests/FunctionalTests/System.Data.SqlClient.Tests.csproj]: See note above IRT UnsupportedPlatform

File: src/System.IO.FileSystem.AccessControl/tests/System.IO.FileSystem.AccessControl.Tests.csproj [hunk 10]
```diff
@@ -16,6 +16,7 @@
     <ProjectReference Include="..\src\System.IO.FileSystem.AccessControl.csproj">
       <Project>{96AA2060-C846-4E56-9509-E8CB9C114C8F}</Project>
       <Name>System.IO.FileSystem.AccessControl</Name>
+      <OSGroup>$(OSEnvironment)</OSGroup>
```
Comment (weshaggard) [inline src/System.IO.FileSystem.AccessControl/tests/System.IO.FileSystem.AccessControl.Tests.csproj]: Should this be `$(TestOsTarget)` like you did in System.IO.Tests?

File: src/System.Net.Http.WinHttpHandler/pkg/System.Net.Http.WinHttpHandler.builds [hunk 11]
```diff
@@ -2,7 +2,9 @@
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <ItemGroup>
-    <Project Include="System.Net.Http.WinHttpHandler.pkgproj" />
```
Comment (weshaggard) [inline src/System.Net.Http.WinHttpHandler/pkg/System.Net.Http.WinHttpHandler.builds]: Why the update the the pkg builds file?

File: src/System.Net.Sockets/src/System.Net.Sockets.csproj [hunk 12]
```diff
@@ -1,8 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <PropertyGroup>
```
Comment (weshaggard) [inline src/System.Net.Sockets/src/System.Net.Sockets.csproj]: Now what happens if you build this csproj locally without having this default? I suspect it fails. We should see about keeping these defaults for local project builds without have to specify the Configuration directly at the command line.

File: src/System.Private.Uri/tests/System.Private.Uri.Tests.csproj [hunk 13]
```diff
@@ -1,25 +1,16 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <PropertyGroup>
-    <Configuration Condition="'$(Configuration)'==''">Windows_Debug</Configuration>
-  </PropertyGroup>
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
-    <Configuration Condition=" '$(Configuration)' == '' ">Windows_Debug</Configuration>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
```
Comment (weshaggard) [inline src/System.Private.Uri/tests/System.Private.Uri.Tests.csproj]: You can remove this Configuration property completely.

File: src/System.Threading.Tasks/src/System.Threading.Tasks.csproj [hunk 14]
```diff
@@ -20,7 +20,7 @@
   <ItemGroup Condition="'$(TargetGroup)' == ''">
     <Compile Include="System\Threading\Tasks\TaskExtensions.CoreCLR.cs" />
     <ProjectReference Include="..\..\System.Diagnostics.Debug\src\System.Diagnostics.Debug.csproj">
-      <OSGroup>Windows_NT</OSGroup>
+      <OSGroup>$(TestOsTarget)</OSGroup>
```
Comment (weshaggard) [inline src/System.Threading.Tasks/src/System.Threading.Tasks.csproj]: TestOSTarget shouldn't be used in the product builds. Is there are reason you needed to update this?

File: src/System.Threading/src/System.Threading.csproj [hunk 15]
```diff
@@ -29,7 +29,7 @@
   </ItemGroup>
   <ItemGroup Condition="'$(TargetGroup)' == ''">
     <ProjectReference Include="..\..\System.Diagnostics.Debug\src\System.Diagnostics.Debug.csproj">
-      <OSGroup>Windows_NT</OSGroup>
+      <OSGroup>$(TestOsTarget)</OSGroup>
```
Comment (weshaggard) [inline src/System.Threading/src/System.Threading.csproj]: TestOSTarget shouldn't be used in the product builds. Is there are reason you needed to update this?

File: src/System.Threading/tests/System.Threading.Tests.csproj [hunk 16]
```diff
@@ -66,12 +63,6 @@
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
       <Targets>Build;DebugSymbolsProjectOutputGroup</Targets>
     </ProjectReference>
-    <ProjectReference Include="..\..\System.Diagnostics.Debug\src\System.Diagnostics.Debug.csproj">
```
Comment (weshaggard) [inline src/System.Threading/tests/System.Threading.Tests.csproj]: Is this reference no longer needed?

File: dir.targets (hunk 2 above)
Comment (sokket) [inline dir.targets]: Looks like I can just use TargetOS instead

Comment (sokket) [inline dir.targets]: When will the BuildTools change be picked up by CoreFx?

Comment (weshaggard) [inline dir.targets]: I think the TargetOS part is already included, but @Priya91 is also working merging another change with https://github.com/dotnet/corefx/pull/7255.

File: src/System.Console/tests/System.Console.Tests.builds (hunk 7 above)
Comment (sokket) [inline src/System.Console/tests/System.Console.Tests.builds]: Fixed

File: src/System.Data.SqlClient/tests/System.Data.SqlClient.Tests.builds (hunk 8 above)
Comment (sokket) [inline src/System.Data.SqlClient/tests/System.Data.SqlClient.Tests.builds]: So this is a work-around for the issue with run-tests where run-tests will fail if a csproj exists but no test binary and without an UnsupportedPlatform tag. Once run-tests is fixed, all the Unsupported tags can go away

File: src/System.IO.FileSystem.AccessControl/pkg/System.IO.FileSystem.AccessControl.builds (hunk 9 above)
Comment (sokket) [inline src/System.IO.FileSystem.AccessControl/pkg/System.IO.FileSystem.AccessControl.builds]: To get around a bin-clash error

File: src/System.IO.FileSystem.AccessControl/tests/System.IO.FileSystem.AccessControl.Tests.csproj (hunk 10 above)
Comment (sokket) [inline src/System.IO.FileSystem.AccessControl/tests/System.IO.FileSystem.AccessControl.Tests.csproj]: Switching to be TargetOS

File: src/System.Net.Sockets/src/System.Net.Sockets.csproj (hunk 12 above)
Comment (sokket) [inline src/System.Net.Sockets/src/System.Net.Sockets.csproj]: Added them back as $(TargetOS)_Debug

File: src/System.Private.Uri/tests/System.Private.Uri.Tests.csproj (hunk 13 above)
Comment (sokket) [inline src/System.Private.Uri/tests/System.Private.Uri.Tests.csproj]: But won't that have the same problem at System.Net.Sockets when building from the command line then?

File: src/System.Threading.Tasks/src/System.Threading.Tasks.csproj (hunk 14 above)
Comment (sokket) [inline src/System.Threading.Tasks/src/System.Threading.Tasks.csproj]: I'll change System.Diagnostics.Debug.csproj's default Configuration to be $(TargetOS)_Debug instead of Windows_Debug to mitigate projects that are targeting AnyOS referencing it and getting the Windows binary

File: src/System.Threading/src/System.Threading.csproj (hunk 15 above)
Comment (sokket) [inline src/System.Threading/src/System.Threading.csproj]: Same as above

File: src/System.Threading/tests/System.Threading.Tests.csproj (hunk 16 above)
Comment (sokket) [inline src/System.Threading/tests/System.Threading.Tests.csproj]: System.Diagnostics.Debug is already referenced in the project.json file

File: src/System.Private.Uri/tests/System.Private.Uri.Tests.csproj (hunk 13 above)
Comment (weshaggard) [inline src/System.Private.Uri/tests/System.Private.Uri.Tests.csproj]: The difference is setting this before and after the import of dir.props. If is set after then it will always be set in dir.props so it will never be used here and dir.props always defaults it to Debug.

File: src/System.Diagnostics.Debug/src/System.Diagnostics.Debug.csproj [hunk 17]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
-    <Configuration Condition="'$(Configuration)'==''">Windows_Debug</Configuration>
+    <Configuration Condition="'$(Configuration)'==''">$(TargetOS)_Debug</Configuration>
```
Comment (weshaggard) [inline src/System.Diagnostics.Debug/src/System.Diagnostics.Debug.csproj]: TargetOS isn't set by the time this is evaluated so you are essentially setting it to `_Debug` which isn't going to help.

File: src/System.Net.Sockets/src/System.Net.Sockets.csproj [hunk 18]
```diff
@@ -6,6 +6,8 @@
     <AssemblyVersion>4.1.0.0</AssemblyVersion>
     <ProjectGuid>{43311AFB-D7C4-4E5A-B1DE-855407F90D1B}</ProjectGuid>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <Configuration Condition=" '$(Configuration)' == '' ">$(TargetOS)_Debug</Configuration>
```
Comment (weshaggard) [inline src/System.Net.Sockets/src/System.Net.Sockets.csproj]: Setting this here isn't going to work as we always set these properties in dir.props.

File: src/System.Private.Uri/tests/System.Private.Uri.Tests.csproj (hunk 13 above)
Comment (sokket) [inline src/System.Private.Uri/tests/System.Private.Uri.Tests.csproj]: Oh okay, removed

File: src/System.Diagnostics.Debug/src/System.Diagnostics.Debug.csproj (hunk 17 above)
Comment (sokket) [inline src/System.Diagnostics.Debug/src/System.Diagnostics.Debug.csproj]: Changed to $(OSEnvironment)

File: src/System.Net.Sockets/src/System.Net.Sockets.csproj (hunk 18 above)
Comment (sokket) [inline src/System.Net.Sockets/src/System.Net.Sockets.csproj]: Moved above dir.props and changed to $(OSEnvironment)

File: src/System.Diagnostics.Debug/src/System.Diagnostics.Debug.csproj [hunk 19]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
-    <Configuration Condition="'$(Configuration)'==''">Windows_Debug</Configuration>
+    <Configuration Condition="'$(Configuration)'==''">$(OSEnvironment)_Debug</Configuration>
```
Comment (weshaggard) [inline src/System.Diagnostics.Debug/src/System.Diagnostics.Debug.csproj]: Can you please test this? I suspect that won't work either.

File: src/System.Net.Sockets/src/System.Net.Sockets.csproj [hunk 20]
```diff
@@ -1,5 +1,9 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">$(OSEnvironment)_Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
```
Comment (weshaggard) [inline src/System.Net.Sockets/src/System.Net.Sockets.csproj]: Platform can be deleted.

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
@dotnet-bot test Innerloop Windows_NT Debug Build and Test

**Comment by sokket** [comment]
@dotnet-bot test  Innerloop CentOS7.1 Debug Build and Test

**Comment by sokket** [comment]
@weshaggard Any further feedback or can I merge this change?

File: build.proj [hunk 21]
```diff
@@ -3,8 +3,11 @@
   <PropertyGroup>
     <!-- Capture OSGroup passed to command line for setting default FilterToOSGroup value below -->
     <_OriginalOSGroup>$(OSGroup)</_OriginalOSGroup>
-    <TestOsTarget Condition="'$(OSGroup)'!=''">$(OSGroup)</TestOsTarget>
-    <TestOsTarget Condition="'$(OSGroup)'==''">$(OSEnvironment)</TestOsTarget>
+  </PropertyGroup>
+  <PropertyGroup>
+    <InputOSGroup>$(OSGroup)</InputOSGroup>
```
Comment (weshaggard) [inline build.proj]: InputOSGroup - Are you trying to use this to capture what is passed at the command line?

File: build.proj [hunk 22]
```diff
@@ -3,8 +3,11 @@
...
+    <InputOSGroup Condition="'$(InputOSGroup)'==''">$(FilterToOSGroup)</InputOSGroup>
+    <InputOSGroup Condition="'$(InputOSGroup)'==''">$(OS)</InputOSGroup>
```
Comment (weshaggard) [inline build.proj]: Should this fallback to OSEnviroment as opposed to OS?

File: src/System.Threading.AccessControl/tests/System.Threading.AccessControl.Tests.csproj [hunk 23]
```diff
@@ -16,7 +16,7 @@
     <ProjectReference Include="..\src\System.Threading.AccessControl.csproj">
       <Project>{E3ED83FD-3015-4BD8-A1B8-6294986E6CFA}</Project>
       <Name>System.Threading.AccessControl</Name>
-      <OSGroup>$(TargetOS)</OSGroup>
+      <OSGroup>$(InputOSGroup)</OSGroup>
```
Comment (weshaggard) [inline src/System.Threading.AccessControl/tests/System.Threading.AccessControl.Tests.csproj]: What happens with this if you build this project in isolation and not from the root build?

**Comment by sokket** [comment]
FYI @maririos

File: build.proj (hunk 21 above)
Comment (sokket) [inline build.proj]: Yes. It will capture whatever is put from the command line OR it will generate it based on the current running platform. This gives me a variable that is guaranteed to be there and have data so I can use it anywhere in the csproj

File: build.proj (hunk 22 above)
Comment (sokket) [inline build.proj]: I thought OSEnvironment was specified by the build.props and build.proj was declared before build.props...is that incorrect?

File: src/System.Threading.AccessControl/tests/System.Threading.AccessControl.Tests.csproj (hunk 23 above)
Comment (sokket) [inline src/System.Threading.AccessControl/tests/System.Threading.AccessControl.Tests.csproj]: Good catch. I added the following in the root dir.props to catch this scenario for all csproj files and provide an InputOSGroup value if one does not exist:

```
<PropertyGroup>
    <InputOSGroup Condition="'$(InputOSGroup)'==''">$(OSEnvironment)</InputOSGroup>
  </PropertyGroup>
```

File: build.proj (hunk 22 above)
Comment (weshaggard) [inline build.proj]: there is no build.props but there is a dir.props which sets OSEnviroment. In general we have tried to use OSEnviroment through all our projects and targets because we define it from OS in dir.props. In general we should use OSEnviroment, unless OS is the only way you can do it like in your default configurations that happen before the import of dir.props.

Comment (weshaggard) [inline build.proj]: Sorry I should have taken a closer look at where we import dir.props in this file. You were using the right one, however with your addition to default InputOSGroup to OSEnviroment in dir.props you can no just remove this fallback here.

File: src/System.Net.Sockets/src/System.Net.Sockets.csproj [hunk 24]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
-    <Configuration Condition="'$(Configuration)'==''">Windows_Debug</Configuration>
+    <Configuration Condition=" '$(Configuration)' == '' ">$(OS)_Debug</Configuration>
```
Comment (weshaggard) [inline src/System.Net.Sockets/src/System.Net.Sockets.csproj]: Lets try this for now but I suspect this isn't going to give us the full experience we want in all scenario.

File: build.proj (hunk 22 above)
Comment (sokket) [inline build.proj]: Removed the OS fallback

File: src/System.Net.Sockets/src/System.Net.Sockets.csproj (hunk 24 above)
Comment (sokket) [inline src/System.Net.Sockets/src/System.Net.Sockets.csproj]: I can probably change that to $(InputOSGroup)...that would give the desired effect

Comment (sokket) [inline src/System.Net.Sockets/src/System.Net.Sockets.csproj]: Ah no it won't, nevermind

Comment (weshaggard) [inline src/System.Net.Sockets/src/System.Net.Sockets.csproj]: Yes it won't but I also don't have a better answer here right now either.

**Comment by weshaggard** [comment]
OK after the removal of the OS fallback please squash and then LGTM 

Thanks for tackling this.

**Comment by sokket** [comment]
Merging. Thanks everyone!
